# Patch to build CoolReader3 with active wxGTK-3.0 profile 
# (eselect wxwidgets is set to "gtk2-unicode-3.0" instead of "gtk2-inicode-release-2.8")
diff -Naur old/cr3wx/src/cr3.cpp new/cr3wx/src/cr3.cpp
--- old/cr3wx/src/cr3.cpp	2015-07-13 11:23:37.000000000 +0300
+++ new/cr3wx/src/cr3.cpp	2017-08-28 12:54:13.000000000 +0300
@@ -395,7 +395,7 @@
 
 lString16 GetConfigFileName()
 {
-    lString16 cfgdir( wxStandardPaths::Get().GetUserDataDir().c_str() );
+    lString16 cfgdir( wxStandardPaths::Get().GetUserDataDir().wx_str() );
     if ( !wxDirExists( cfgdir.c_str() ) )
         ::wxMkdir( wxString( cfgdir.c_str() ) );
     lChar16 slash = detectSlash( cfgdir );
@@ -622,7 +622,7 @@
     wxImage::AddHandler(new wxPNGHandler);
     resources = new ResourceContainer();
 
-    lString16 appname( argv[0] );
+    lString16 appname( argv[0].wx_str() );
     int lastSlash=-1;
     lChar16 slashChar = '/';
     for ( int p=0; p<(int)appname.length(); p++ ) {
@@ -777,7 +777,7 @@
     int argc = wxGetApp().argc;
     lString16 fnameToOpen;
     for ( int i=1; i<argc; i++ ) {
-        lString16 param = lString16( wxGetApp().argv[1] );
+        lString16 param = lString16( wxGetApp().argv[1].wx_str() );
         if ( param[0]!='-' )
             fnameToOpen = param;
     }
@@ -1193,7 +1193,7 @@
     lString16 outFile;
     bool convert = false;
     for ( int i=1; i<argc; i++ ) {
-        lString16 param = lString16( wxGetApp().argv[i] );
+        lString16 param = lString16( wxGetApp().argv[i].wx_str() );
         if ( param[0]!='-' )
             fnameToOpen = param;
         else if (param.startsWith("--convert"))
@@ -1470,7 +1470,7 @@
         wxCursor hg( wxCURSOR_WAIT );
         this->SetCursor( hg );
         wxSetCursor( hg );
-        _view->getDocView()->exportWolFile( dlg.GetPath(), opts.getMode()==0, opts.getLevels() );
+        _view->getDocView()->exportWolFile( dlg.GetPath().wx_str(), opts.getMode()==0, opts.getLevels() );
         wxSetCursor( wxNullCursor );
         this->SetCursor( wxNullCursor );
     }
diff -Naur old/cr3wx/src/optdlg.cpp new/cr3wx/src/optdlg.cpp
--- old/cr3wx/src/optdlg.cpp	2015-07-13 11:23:37.000000000 +0300
+++ new/cr3wx/src/optdlg.cpp	2017-08-28 11:38:06.000000000 +0300
@@ -41,7 +41,7 @@
             if ( v==_choices[i] )
                 tb = i;
         if ( _storeStringValues ) {
-            props->setString( _option, lString16(_choices[tb]) );
+            props->setString( _option, lString16(_choices[tb].wx_str()) );
         } else {
             props->setInt( _option, tb );
         }
@@ -50,7 +50,7 @@
     {
         unsigned tb = _defvalue;
         if ( _storeStringValues ) {
-            lString8 s8 = UnicodeToUtf8( lString16(_choices[_defvalue]) );
+            lString8 s8 = UnicodeToUtf8( lString16(_choices[_defvalue].wx_str()) );
             lString16 s16 = props->getStringDef( _option, s8.c_str() );
             wxString v = s16.c_str();
             for ( unsigned i=0; i<_choices.GetCount(); i++ )
diff -Naur old/cr3wx/src/view.cpp new/cr3wx/src/view.cpp
--- old/cr3wx/src/view.cpp	2015-07-13 11:23:37.000000000 +0300
+++ new/cr3wx/src/view.cpp	2017-08-28 12:57:55.000000000 +0300
@@ -353,7 +353,7 @@
 
 lString16 cr3view::GetHistoryFileName()
 {
-    lString16 cfgdir( wxStandardPaths::Get().GetUserDataDir().c_str() );
+    lString16 cfgdir( wxStandardPaths::Get().GetUserDataDir().wx_str() );
     if ( !wxDirExists( cfgdir.c_str() ) )
         ::wxMkdir( wxString( cfgdir.c_str() ) );
     lChar16 slash = detectSlash( cfgdir );
@@ -667,13 +667,13 @@
     //===========================================
     GetParent()->Update();
     //printf("   loading...  ");
-    bool res = getDocView()->LoadDocument( fname.c_str() );
+    bool res = getDocView()->LoadDocument( fname.wx_str() );
     //printf("   done. \n");
 	//DEBUG
 	//_docview->exportWolFile( "test.wol", true );
 	//_docview->SetPos(0);
     if ( !res )
-        getDocView()->createDefaultDocument(lString16("File open error"), lString16("Cannot open file ") + fname.c_str() );
+        getDocView()->createDefaultDocument(lString16("File open error"), lString16("Cannot open file ") + fname.wx_str() );
     lString16 title = getDocView()->getAuthors();
     if ( !title.empty() && !getDocView()->getTitle().empty() )
         title << L". ";
